apis/operator/v1alpha1/discoveryservice_types.go

```diff
@@ -76,6 +82,17 @@ type DiscoveryService struct {
 	Status DiscoveryServiceStatus `json:"status,omitempty"`
 }
 
+func (d *DiscoveryService) Resources() v1.ResourceRequirements {
```

Comment:
This has been implemented at the API types level.

Another alternative was to do it in the deployment reconciler but then that reconciler would be specific to DiscoveryService.
Another alternative is that the deployment reconciler got the resources as a parameter but it didn't seem very clean.

Reply:
I like this approach. Better to leave the deployment reconciler completely generic as we will be using it also for the Ingress implementation that we plan to do.